orchagent/portsorch.cpp

```diff
@@ -908,59 +935,57 @@ bool PortsOrch::setPortPfcAsym(Port &port, string pfc_asym)
     SWSS_LOG_ENTER();
 
     sai_attribute_t attr;
-    uint8_t pfc = 0;
+    uint8_t pfc_tx_bitmask = 0;
+    uint8_t pfc_rx_bitmask = 0;
 
-    if (!getPortPfc(port.m_port_id, &pfc))
+    if (pfc_asym_map.find(pfc_asym) == pfc_asym_map.end())
     {
+        SWSS_LOG_ERROR("Incorrect asymmetric PFC mode: %s", pfc_asym.c_str());
         return false;
     }
 
-    auto found = pfc_asym_map.find(pfc_asym);
-    if (found == pfc_asym_map.end())
+    if (port.m_pfc_info.pfc_mode == pfc_asym_map[pfc_asym])
+    {
+        SWSS_LOG_NOTICE("Asymmetric PFC mode is alredy set to \"%s\" for the port 0x%" PRIx64,
+            pfc_asym.c_str(), port.m_port_id);
+        return true;
+    }
+
+    if (!getPortPfc(port.m_port_id, &pfc_tx_bitmask, &pfc_rx_bitmask))
     {
-        SWSS_LOG_ERROR("Incorrect asymmetric PFC mode: %s", pfc_asym.c_str());
         return false;
     }
 
-    auto new_pfc_asym = found->second;
-    if (port.m_pfc_asym == new_pfc_asym)
+    if (pfc_asym_map[pfc_asym] == SAI_PORT_PRIORITY_FLOW_CONTROL_MODE_SEPARATE)
     {
-        SWSS_LOG_NOTICE("Already set asymmetric PFC mode: %s", pfc_asym.c_str());
-        return true;
+        pfc_rx_bitmask = static_cast<uint8_t>(0xff);
+    }
+    else
+    {
+        pfc_rx_bitmask = pfc_tx_bitmask;
```

Comment:
logic has changed, can you elaborate it in the code?

Reply:
Logic is exactly the same but code was refactored according to changes for set/getPortPfc API.